### ISSUE TITLE
Use 'Remove' and 'Delete' properly

### ISSUE THREAD

**Issue by RalfKoban** [RalfKoban/MiKo-Analyzers#442]
The terms `Delete` and `Remove` are defined quite similarly.

However, the main difference between them is:
- `delete` means `erase` (i.e. rendered nonexistent or nonrecoverable)
- `remove` means `take away and set aside` (but kept in existence)

So we should not use both terms together in documentation and naming (e.g. of methods).

For example, following code is misleading and should trigger a warning:
```c#
/// <summary>
/// Deletes the given stuff.
/// </summary>
void Remove(Stuff stuff);
```

It is misleading (or can be misinterpreted) because the question arises:
**_Is the stuff non-recoverably erased?
Or is it still valid stuff, but simply taken away from the specific instance (e.g. from a list)?_**
